src/main/java/com/epam/lowcost/controller/UserController.java

```diff
@@ -5,37 +5,56 @@
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Controller;
 import org.springframework.ui.Model;
+import org.springframework.ui.ModelMap;
 import org.springframework.web.bind.annotation.*;
 
 import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
 import java.util.Map;
 
 import static com.epam.lowcost.util.EndPoints.*;
 
 @Controller
 @RequestMapping(value = USER)
-@SessionAttributes(value = "sessionUser")
+@SessionAttributes({"sessionUser", "number"})
 public class UserController {
 
     @Autowired
     UserService userService;
 
 
-    @GetMapping(value = ALL)
-    public String getAllUsers(@ModelAttribute(value = "sessionUser") User sessionUser, Model model) {
+    @GetMapping(value = "all/{pageId}")
+    public String getAllUsers(@PathVariable int pageId, @ModelAttribute(value = "sessionUser") User sessionUser, ModelMap model) {
         if (!sessionUser.isAdmin()) {
             return "redirect:" + TICKETS + SELF;
         }
-        model.addAttribute("users", userService.getAllUsers());
+
+        int usersByPage = (int) model.getOrDefault("number", 5);
+
+        Map<String, Object> pageRepresentation = userService.getUsersByPage(pageId, usersByPage);
+
+        model.addAttribute("pagesNum", pageRepresentation.get("pagesNum"));
+        model.addAttribute("users", pageRepresentation.get("users"));
+        model.addAttribute("pageId", pageRepresentation.get("pageId"));
         return "users";
     }
 
+    @GetMapping(value = "/setUsersByPage")
+    public String setUsersByPage(@RequestParam String number, Model model) {
+        model.addAttribute("number", Integer.parseInt(number));
+        return "redirect:/user/all/1";
+    }
+
+
     @GetMapping
     public String getById(@ModelAttribute(value = "sessionUser") User sessionUser, @RequestParam long id, Model model) {
         if (!sessionUser.isAdmin()) {
             return "redirect:" + TICKETS + SELF;
         }
-        model.addAttribute("user", userService.getById(id));
+        List<User> user = new ArrayList<>();
+        user.add(userService.getById(id));
+        model.addAttribute("users", user);
         model.addAttribute("message", "Here is your User!");
         return "users";
```

Comment:
?

Reply:
Object we put to model should be a Collection in order to be iterable in same <c:forEach/> as every other returns from UserController.  Could be changed to userService.getById returning List<User>.